Documentation/app-operations.md

```diff
@@ -0,0 +1,241 @@
+# Fine-Grained Pod Manipulations
+
+To provide imperative application capabilities (e.g. start|stop|inject) inside a
+pod for finer-grained control over rkt containerization concepts and debugging
+needs, this proposal introduces new stage1 entrypoints that will be used for
+manipulating applications inside pods.
+
+The motivation behind this change is the direction orchestration systems are
+taking and to facilitate this new direction. For more details, see
+[kubernetes#25899](https://github.com/yujuhong/kubernetes/blob/08dc66113399c89e31f6872f3c638695a6ec6a8d/docs/proposals/container-runtime-interface-v1.md).
+
+# API
+
+The envisioned workflow for the app-level CLI API is that After a pod has been started, 
+the users will be invoking the rkt CLI which is application logic on top of stage1 entrypoints (introduced below).
+
+## rkt app inject
+Injects an application image into a running pod.
+
+It first prepares an application rootfs in stage0 for the application image/images
+and then the prepared-app is injected via the entrypoint.
+
+```bash
+rkt app inject <pod-uuid> <image-name/hash/address/registry-URL> <arguments>
+```
+
+## rkt app eject
+Ejects an application image into a running pod, meaning that the leftover resources are removed from the pod.
```

Comment:
@yifan-gu the use case I had in mind was when deleting a pod while there are apps running inside it. In which case `app eject` would be invoked for each app running or not